documentation/library_guide/extension_api.rst

```diff
@@ -255,17 +256,20 @@ These algorithms are declared in ``oneapi::dpl::experimental::ranges`` namespace
 In order to make these algorithm available the ``<oneapi/dpl/ranges>`` header should be included.
 Use of the range-based API requires C++17 and the C++ standard libraries coming with GCC 8.1 (or higher) or Clang 7 (or higher).
 
-The following viewable ranges are declared in ``oneapi::dpl::experimental::ranges`` namespace. Only those are allowed to use as ranges
+The following views are declared in the ``oneapi::dpl::experimental::ranges`` namespace. Only those are allowed to use as ranges
 for range-based algorithms.
 
 * ``iota_view``: A range factory - generates a sequence of N elements which starts from an initial value and ends by final N-1.
+* ``generate``:  A range factory - generates a sequence of N elements, where each is produced by a given functional genrator.
+* ``fill ``: A range factory - generates a sequence of N elements, where each is equal a given value.
 * ``all_view``: A custom utility - represents a view of all or a part of ``sycl::buffer`` underlying elements.
 * ``guard_view``: A custom utility - represents a view of USM data range defined by a two USM pointers.
 * ``zip_view``: A custom range adapter - produces one ``zip_view`` from other several views.
 * ``transform_view``: A range adapter - represents a view of a underlying sequence after applying a transformation to each element.
 * ``reverse_view``: A range adapter - produces a reversed sequence of elements provided by another view.
 * ``take_view``: A range adapter - produces a view of the first N elements from another view.
 * ``drop_view``: A range adapter - produces a view excluding the first N elements from another view.
+* ``rotate``: A range adapter - produces a left rotated sequence of elements provided by another view.
 
```

Comment:
Just a suggestion.
Basing on description of views I see that the list can be split to 3 sections: factories, utilities, adapters. Since the list is growing there will be more elements in this list. Splitting into sections can help users find an information. Will it be better if we do such splitting here?

Reply:
Probably, it makes sense...
But I think that "reordering" is enough - "utilities, factories, adapters". 
Otherwise, in case of split we need introduce the sections  "utilities, factories, adapters" explicitly. In that case we have to  give definitions of ones terms...  But I would not  introduce additional definitions from C++ standard here..

Reply:
I agree with Mike, and like the reordering option. If we do go with splitting to "utilities, factories, adapters", I think having a more extensive bullet list would fix the issue:

- Adapters:
    - Adapter 1
    - Adapter 2
- Factories:
    - Factory 1
    - Factory 2
- Utilities:
    - Utility 1
    - Utility 2